sdk/monitor/monitor-opentelemetry/src/traces/sampler.ts

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { Link, Attributes, SpanKind, Context } from "@opentelemetry/api";
+import { Sampler, SamplingDecision, SamplingResult } from "@opentelemetry/sdk-trace-base";
+
+/**
+ * ApplicationInsightsSampler is responsible for the following:
+ * Implements same trace id hashing algorithm so that traces are sampled the same across multiple nodes
+ * Adds item count to span attribute if span is sampled (needed for ingestion service)
+ * @param samplingRatio - 0 to 1 value.
+ */
+export class ApplicationInsightsSampler implements Sampler {
+  private readonly _sampleRate: number;
+  private readonly _samplingRatio: number;
+
+  /**
+   * Initializes a new instance of the ApplicationInsightsSampler class.
+   * @param samplingRatio Value in the range [0,1], 1 meaning all data will sampled and 0 all Tracing data will be sampled out.
+   */
+  constructor(samplingRatio: number = 1) {
+    this._samplingRatio = samplingRatio;
+    if (this._samplingRatio > 1) {
+      throw new Error("Wrong sampling rate, data will not be sampled out");
+    }
+    this._sampleRate = Math.round(this._samplingRatio * 100);
+  }
+
+  /**
+   * Checks whether span needs to be created and tracked.
+   *
+   * @param context Parent Context which may contain a span.
+   * @param traceId of the span to be created. It can be different from the
+   *     traceId in the {@link SpanContext}. Typically in situations when the
+   *     span to be created starts a new trace.
+   * @param spanName of the span to be created.
+   * @param spanKind of the span to be created.
+   * @param attributes Initial set of SpanAttributes for the Span being constructed.
+   * @param links Collection of links that will be associated with the Span to
+   *     be created. Typically useful for batch operations.
+   * @returns a {@link SamplingResult}.
+   */
+  public shouldSample(
+    // @ts-ignore
```

Comment:
What was the typing issue that lead to all these `@ts-ignore`s? A short comment about why typing isn't working here and why we're ignoring could be helpful.

Reply:
Unused parameters, OpenTelemetry API define these but we are not using them in our code